src/awscr-s3/client.cr

```diff
@@ -70,8 +70,8 @@ module Awscr::S3
       resp.status_code == 204
     end
 
-    def put_object(bucket, object : String, body : IO | String)
-      resp = http.put("/#{bucket}/#{object}", body)
+    def put_object(bucket, object : String, body : IO | String, headers : Hash(String, String) = Hash(String, String).new)
```

Comment:
Maybe change `headers` to metadata?

Reply:
I think that if we were to only allow metadata (like append `x-aws-meta-` everywhere, thus simplifying things a bit), that would be appropriate, but then again these can be also `x-aws-server-` or `x-aws-storage-` or other things in DO (`Content-Type`, `Content-Disposition`). I'm fine with either but I think `headers` is a bit more explicit.

Reply:
on the other hand, we do set `Content-Length` ourselves so `metadata` or `additional_headers` would also fit...